spec/acceptance/class_spec.rb

```diff
@@ -65,6 +65,42 @@ class { 'nodejs':
     end
   end
 
+  context 'RedHat with repo_class => nodejs::repo::dnfmodule', if: fact('os.family') == 'RedHat' && %w[8 9].include?(fact('os.release.major')) do
+    include_examples 'cleanup'
+
+    # Node 20 is only available in Stream yet, not in a released EL
+    # So we're testing 18 here
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'nodejs':
+          nodejs_dev_package_ensure => installed,
+          npm_package_ensure        => installed,
+          repo_class                => 'nodejs::repo::dnfmodule',
+          repo_version              => '18',
+        }
+        PUPPET
+      end
+    end
+
+    %w[
+      npm
+      nodejs
+      nodejs-devel
+    ].each do |pkg|
+      describe package(pkg) do
+        it do
+          is_expected.to be_installed
+        end
+
+        it 'comes from the expected source' do
```

Comment:
Add a `it 'has the right version'` that asserts the output of `shell('node --version')` to be match `v18` (or well, whatever we install)